splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Performance_IT.java

```diff
@@ -0,0 +1,159 @@
+
+		spliceClassWatcher.execute(format("call syscs_util.syscs_flush_table('%s', 'sourceTable')", CLASS_NAME));
+
+        sqlText = "CREATE TRIGGER Trig%d\n" +
+                "  NO CASCADE\n" +
+                "  BEFORE INSERT ON targetTable\n" +
+                "  REFERENCING NEW AS N\n" +
+                "  FOR EACH ROW\n" +
+                "  WHEN ((N.a NOT IN\n" +
+                "         (SELECT A.a FROM dummyTable A, dummyTable B WHERE A.a = B.a ))\n" +
+                "        AND\n" +
+                "        (N.b NOT IN (SELECT b FROM dummyTable C WHERE C.a = 2)))\n" +
+                "  BEGIN ATOMIC\n" +
+                "    select 1;\n" +
+                "  END";
+        for (int i=0; i < 15; i++)
+            spliceClassWatcher.executeUpdate(format(sqlText, i));
+    }
+
+    @BeforeClass
+    public static void recordNumTables() throws Exception {
+        isMemPlatform = isMemPlatform(spliceClassWatcher);
+        vacuum();
+        numTables = getNumTables();
+    }
+
+    @AfterClass
+    public static void checkNumTables() throws Exception {
+        spliceSchemaWatcher.cleanSchemaObjects();
+        spliceClassWatcher.close();
+        vacuum();
+        int newNumTables = getNumTables();
+        // Mem platform doesn't physically remove dropped tables, so
+        // this is an HBase-only check.
+        if (!isMemPlatform)
+            assertTrue("\nStarted with " + numTables + " tables and ended with " + newNumTables,
+                         numTables >= newNumTables);
+    }
+
+    public static void
+    vacuum() throws Exception{
+        // Mem doesn't have vacuum.
+        if (!isMemPlatform)
+            spliceClassWatcher.executeUpdate("CALL SYSCS_UTIL.VACUUM()");
+    }
+
+    public static int
+    getNumTables() throws Exception{
+        try (ResultSet rs = spliceClassWatcher.executeQuery("CALL SYSCS_UTIL.SYSCS_GET_TABLE_COUNT()")) {
+            rs.next();
+            return ((Integer)rs.getObject(1));
+        }
+    }
+
+    @Test
+    public void testConcurrentTriggersRuntimeUnderLimit() throws Exception {
+        // Make sure spark is warmed up
+        if (useSpark) {
+            try (ResultSet rs = methodWatcher.executeQuery("select * from sourceTable" +
+                "--splice-properties useSpark=" + useSpark + "\n")) {
+            }
+            try (ResultSet rs = methodWatcher.executeQuery("select * from sourceTable" +
+                "--splice-properties useSpark=" + useSpark + "\n")) {
+            }
+        }
+        long startTime = System.currentTimeMillis();
+        methodWatcher.execute("insert into targetTable --splice-properties useSpark=" + useSpark +
+                              "\n select * from sourceTable");
+        long endTime = System.currentTimeMillis();
+        long runTime = endTime - startTime;
+        assertTrue("Expected runtime to be less than 10 seconds.  Actual time: " + runTime + " milliseconds", runTime < 10000);
+
```

Comment:
This might be a bit flaky when running on jenkins, hopefully it works well. I think we need a different platform to track this kind of perf regressions.

Reply:
OK.  If need be I can bump the check up to a higher timeout.